wire-ios-data-model/Tests/OneOnOne/OneOnOneResolverTests.swift

```diff
@@ -95,16 +221,52 @@ final class OneOnOneResolverTests: ZMBaseManagedObjectTest {
         }
 
         // Mock
-        protocolSelector.getProtocolForUserWithIn_MockValue = .some(nil)
+        await mockProtocolSelector.setGetProtocolForUserWithIn_MockValue(.some(nil))
 
         // When
-        try await sut.resolveOneOnOneConversation(with: userID, in: uiMOC)
+        try await sut.resolveOneOnOneConversation(with: userID, in: viewContext)
 
         // Then
-        await uiMOC.perform {
+        await viewContext.perform {
             XCTAssertEqual(conversation.messageProtocol, .proteus)
             XCTAssertTrue(conversation.isForcedReadOnly)
         }
     }
 
+    // MARK: - Helpers
+
+    @discardableResult
+    private func createUser(in context: NSManagedObjectContext) -> ZMUser {
+        let user = ZMUser.insertNewObject(in: context)
+        user.remoteIdentifier = UUID()
+        user.domain = "local@domain.com"
+        return user
+    }
+
+    @discardableResult
+    private func createConnection(
+        status: ZMConnectionStatus,
+        to user: ZMUser,
+        in context: NSManagedObjectContext
+    ) -> (ZMConnection, ZMConversation) {
+        let connection = ZMConnection.insertNewObject(in: context)
+        connection.to = user
+        connection.status = status
+        connection.message = "Connect to me"
+        connection.lastUpdateDate = .now
+
+        let conversation = ZMConversation.insertNewObject(in: context)
+        conversation.conversationType = .connection
+        conversation.remoteIdentifier = UUID()
+        conversation.domain = "local@domain.com"
+        user.oneOnOneConversation = conversation
+
+        return (connection, conversation)
```

Comment:
non-blocking: this could be moved to the `ModelHelper`.

Reply:
Thanks, will make a follow up to this, because all tests passed and it's mergeable now.